src/content/components/TimelineCanvas.js

```diff
@@ -3,33 +3,36 @@ import React, { Component } from 'react';
 import { timeCode } from '../../common/time-code';
 import classNames from 'classnames';
 
-import type { CssPixels, DevicePixels, NonNull } from '../../common/types/units';
+import type { CssPixels, DevicePixels } from '../../common/types/units';
 
-type HoveredItem = NonNull;
-
-type Props = {
+type Props<HoveredItem> = {
   containerWidth: CssPixels,
   containerHeight: CssPixels,
   className: string,
-  onDoubleClickItem: HoveredItem => void,
+  onDoubleClickItem: (HoveredItem | null) => void,
   getHoveredItemInfo: HoveredItem => string,
-  drawCanvas: (CanvasRenderingContext2D, HoveredItem) => void,
-  hitTest: (x: CssPixels, y: CssPixels) => null | HoveredItem,
+  drawCanvas: (CanvasRenderingContext2D, HoveredItem | null) => void,
+  hitTest: (x: CssPixels, y: CssPixels) => HoveredItem | null,
 };
 
-require('./TimelineCanvas.css');
+type State<HoveredItem> = {
+  hoveredItem: HoveredItem | null,
+};
 
-export default class TimelineCanvas extends Component {
+require('./TimelineCanvas.css');
 
-  props: Props;
+export default class TimelineCanvas<HoveredItem> extends Component<
+  void,
+  Props<HoveredItem>,
+  State<HoveredItem>
+> {
```

Comment:
I couldn't get flow to pass without doing this. I even broke it down into a simplified test case (which I've already deleted apparently)